File: src/main/java/vergisst/minecraftmod/weaponthrow/events/OnFOVUpdate.java
package vergisst.minecraftmod.weaponthrow.events;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.entity.player.PlayerEntity;

public interface OnFOVUpdate {
    Event<OnFOVUpdate> EVENT = EventFactory.createArrayBacked(OnFOVUpdate.class,
            (listeners) -> (player, fov) -> {
                var currentFov = fov;
                for (var listener: listeners) {
                    currentFov = listener.interact(player, fov);
                }

                return currentFov;
            });
    float interact(PlayerEntity entity, float fov);
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/events/OnFOVUpdate.kt
package vergisst.minecraftmod.weaponthrowlite.events

import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.entity.player.PlayerEntity

interface OnFOVUpdate {
    fun interact(player: PlayerEntity, fov: Float): Float

    val EVENT
        get() = EventFactory.createArrayBacked(
        OnFOVUpdate::class.java,
        {listeners ->
            object: OnFOVUpdate {
                override fun interact(player: PlayerEntity, fov: Float): Float {
                    for (item in listeners) {
                        val result = item.interact(player, fov)

                        if (result != 0f) {
                            return result
                        }
                    }
                    return 0f
                }
            }
        })

}